astroid/inference.py

```diff
@@ -130,7 +131,10 @@ def infer_map(self, context=None):
         yield new_seq
 
 
-def _update_with_replacement(lhs_dict, rhs_dict):
+def _update_with_replacement(
+    lhs_dict: dict[InferenceResult, InferenceResult],
```

Comment:
@cdce8p I know you previously said you don't like using this alias in argument annotations, but I wonder if we should reconsider that.

I don't think it makes a lot of sense to create the alias in every module that uses inference results and it also helps a bit with readability if we just use it consistently.
We could also import the alias in `pylint` and build its typing on top of `astroid's`

Reply:
I agree, the alias does make sense here.

Reply:
Follow up:
Do we want to start importing this in `pylint` as well? Or do we still consider `astroid/typing.py` private and define `InferenceResult` in `pylint/typing.py` as well?
I'm not sure how `mypy` and `pyright` handle this. Are they able to infer that `pylint.InferenceResult` and `astroid.InferenceResult` would be the same?

Reply:
> Do we want to start importing this in `pylint` as well?

For `safe_infer`?

> Or do we still consider `astroid/typing.py` private and define `InferenceResult` in `pylint/typing.py` as well?

I would recommend it pre se, but `InferenceResult` should be fine.

> I'm not sure how `mypy` and `pyright` handle this. Are they able to infer that `pylint.InferenceResult` and `astroid.InferenceResult` would be the same?

They should be, once we have added `py.typed` to astroid at least. Type checkers will resolve the union and compare each type individually.

Reply:
> For `safe_infer`?

Yeah and in other places. There are methods which get an `inferred` from their callee, which might require the full scope of `InferenceResult` because they haven't been narrowed down before the call.
 
> I would recommend it pre se, but `InferenceResult` should be fine.

Let's redefine. Saves potential headaches. I think with the addition of `Proxy` we have covered all case so we shouldn't need to update it.

> They should be, once we have added `py.typed` to astroid at least. Type checkers will resolve the union and compare each type individually.

once... 😄